website/addons/box/model.py

```diff
@@ -1,28 +1,59 @@
 # -*- coding: utf-8 -*-
-import time
 import logging
-from datetime import datetime
 
-import furl
 import pymongo
-import requests
-from modularodm import fields, StoredObject
-from box import CredentialsV2, refresh_v2_token, BoxClientException
+from box import CredentialsV2, BoxClient
+from box.client import BoxClientException
+from modularodm import fields
 
 from framework.auth import Auth
 from framework.exceptions import HTTPError
 
 from website.addons.base import exceptions
-from website.addons.base import AddonUserSettingsBase, AddonNodeSettingsBase, GuidFile
+from website.addons.base import AddonOAuthUserSettingsBase, AddonOAuthNodeSettingsBase, GuidFile
 
 from website.addons.box import settings
-from website.addons.box.utils import BoxNodeLogger
-from website.addons.box.client import get_client_from_user_settings
-
+from website.addons.box.utils import BoxNodeLogger, refresh_oauth_key
+from website.addons.box.serializer import BoxSerializer
+from website.oauth.models import ExternalProvider
 
 logger = logging.getLogger(__name__)
 
 
+class Box(ExternalProvider):
+    name = 'Box'
+    short_name = 'box'
+
+    client_id = settings.BOX_KEY
+    client_secret = settings.BOX_SECRET
+
+    auth_url_base = settings.BOX_OAUTH_AUTH_ENDPOINT
+    auto_refresh_url = settings.BOX_OAUTH_TOKEN_ENDPOINT
+    default_scopes = ['root_readwrite']
+
+    def handle_callback(self, response):
+        """View called when the Oauth flow is completed. Adds a new BoxUserSettings
+        record to the user and saves the user's access token and account info.
+        """
+
+        client = BoxClient(CredentialsV2(
+            response['access_token'],
+            response['refresh_token'],
+            settings.BOX_KEY,
+            settings.BOX_SECRET,
+        ))
+
+        about = client.get_user_info()
+
+        url = 'https://app.box.com/profile/' + about['id']
```

Comment:
Not a big deal, but [new-style string formatting is preferred](http://cosdev.readthedocs.org/en/latest/style_guides/python.html#string-formatting).